libs/compare.go

```diff
@@ -0,0 +1,31 @@
+package libs
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+)
+
+func StreamsEqual(s1 io.Reader, s2 io.Reader) (bool, error) {
+	const chunkSize = 64000
+	b1 := make([]byte, chunkSize)
+	b2 := make([]byte, chunkSize)
+	for {
+		_, err1 := s1.Read(b1)
```

Comment:
It may be not relevant for our simple usage, but this can fail in some cases. This code have an assumption that `Read()` always reads exactly `len(b1)` bytes (or less if EOF), but it may not. See doc for `io.Reader`. You could probably use some helpers like `io.ReadFull` for this.

You can reproduce this error if wrap `bytes.NewReader` in `iotest.HalfReader` or `iotest.OneByteReader`.

Simple TODO with description would be enough for now, but I thought you'd find it interesting.

Reply:
You were right, it was interesting!

Reply:
it would be nice to have tests for this, at least one simple with `iotest.OneByteReader`